File: lab/python/matematica_e_operadores/maior_valor_e_indice.py
def versao1():
    lista = []

    for _ in range(8):
        numero = int(input())
        lista.append(numero)

    maior = lista[0]
    maior_indice = 0

    for i, valor in enumerate(lista):
        if valor > maior:
            maior_indice = i
            maior = valor

    print(f"max={maior}; idx={maior_indice}")


def versao2():
    lista = [int(input()) for _ in range(8)]
    maior = max(lista)
    maior_indice = lista.index(maior)
    print(f"max={maior}; idx={maior_indice}")


versao2()

File: lab/c/matematica_e_operadores/maior_valor_e_indice.c
#include <stdio.h>


void versao1();


int main(){
    versao1();
    return 0;
}


void versao1(){
    int tam = 8;

    int array[tam];

    for (int i=0; i < tam; i++){
        scanf("%d", &array[i]);
    }

    int maior = array[0];
    int maior_indice = 0;

    for (int i=0; i < tam; i++){
        if (array[i] > maior){
            maior_indice = i;
            maior = array[i];
        }
    }

    printf("max=%d; idx=%d\n", maior, maior_indice);
}
